src/main/java/org/fxmisc/flowless/VirtualFlow.java

```diff
@@ -418,4 +448,62 @@ private void jumpToAbsolutePosition(double pixels) {
             navigator.setTargetPosition(new EndOffEnd(items.size() - 1, 0.0));
         }
     }
+
+    /**
+     * The gravity of the virtual flow.  When there are not enough cells to fill
+     * the full height (vertical virtual flow) or width (horizontal virtual flow),
+     * the cells are placed either at the front (vertical: top, horizontal: left),
+     * or rear (vertical: bottom, horizontal: right) of the virtual flow, depending
+     * on the value of the gravity property.
+     *
+     * The gravity can also be styled in CSS, using the "-fx-gravity" property,
+     * for example:
+     * <pre>.virtual-flow { -fx-gravity: rear; }</pre>
+     */
+    public ObjectProperty<Gravity> gravityProperty()
+    {
+        return gravity;
+    }
+
+    public Gravity getGravity()
+    {
+        return gravity.get();
+    }
+
+    public void setGravity(Gravity gravity)
+    {
+        this.gravity.set(gravity);
+    }
+
+    @SuppressWarnings("unchecked") // Because of the cast we have to perform, below
+    private static final CssMetaData<VirtualFlow, Gravity> GRAVITY = new CssMetaData<VirtualFlow, Gravity>(
+            "-fx-gravity",
+            // JavaFX seems to have an odd return type on getEnumConverter: "? extends Enum<?>", not E as the second generic type.
+            // Even though if you look at the source, the EnumConverter type it uses does have the type E.
+            // To get round this, we cast on return:
+            (StyleConverter<?, Gravity>) StyleConverter.getEnumConverter(Gravity.class),
+            Gravity.FRONT) {
+
+        @Override
+        public boolean isSettable(VirtualFlow virtualFlow) {
+            return !virtualFlow.gravity.isBound();
+        }
+
+        @Override
+        public StyleableProperty<Gravity> getStyleableProperty(VirtualFlow virtualFlow) {
+            return virtualFlow.gravity;
+        }
+    };
+
+    @Override
+    public List<CssMetaData<? extends Styleable, ?>> getCssMetaData() {
```

Comment:
There should also be another method `public static getClassCssMetaData()` that returns the same static list